core/src/main/java/org/mule/runtime/core/policy/CompositeSourcePolicy.java

```diff
@@ -116,34 +119,35 @@ protected Event processPolicy(Policy policy, Processor nextProcessor, Event even
    * {@link SourcePolicyParametersTransformer} then those parameters will be exactly the one defined by the message source.
    * 
    * @param sourceEvent the event generated from the source.
-   * @return a {@link SuccessSourcePolicyResult} which contains the response parameters and the result event of the execution or a
-   *         {@link FailureSourcePolicyResult} which contains the failure response parameters and the {@link MessagingException}
+   * @return a {@link SourcePolicySuccessResult} which contains the response parameters and the result event of the execution or a
+   *         {@link SourcePolicyFailureResult} which contains the failure response parameters and the {@link MessagingException}
    *         thrown by the policy chain execution.
    * @throws Exception if there was an unexpected failure thrown by executing the chain.
    */
   @Override
-  public Either<FailureSourcePolicyResult, SuccessSourcePolicyResult> process(Event sourceEvent) throws Exception {
-    try {
-      Event policiesResultEvent = processPolicies(sourceEvent);
-      Map<String, Object> responseParameters =
-          getParametersTransformer().map(parametersTransformer -> concatMaps(originalResponseParameters, parametersTransformer
-              .fromMessageToSuccessResponseParameters(policiesResultEvent.getMessage()))).orElse(originalResponseParameters);
-      return right(new SuccessSourcePolicyResult(policiesResultEvent, responseParameters, getParametersProcessor()));
-    } catch (FlowExecutionException e) {
-      Map<String, Object> responseParameters =
-          getParametersTransformer()
+  public Publisher<Either<SourcePolicyFailureResult, SourcePolicySuccessResult>> process(Event sourceEvent) {
```

Comment:
clarify this code by either adding comments or translating it to English from Arameic